retrofit-mock/src/main/java/retrofit/MockRetrofit.java

```diff
@@ -0,0 +1,182 @@
+  private static final int DEFAULT_VARIANCE_PCT = 40; // Network delay varies by ±40%.
+  private static final int DEFAULT_ERROR_PCT = 3; // 3% of network calls will fail.
+  private static final int DEFAULT_ERROR_DELAY_FACTOR = 3; // Errors will be scaled by this value.
+
+  /**
+   * Create an instance with a normal {@link Retrofit} instance and an executor service on which
+   * the simulated delays will be created. Instances of this class should be re-used so that the
+   * behavior of every mock service is consistent.
+   */
+  public static MockRetrofit from(Retrofit restAdapter, ExecutorService backgroundExecutor) {
+    return new MockRetrofit(restAdapter, backgroundExecutor);
+  }
+
+  private final ExecutorService backgroundExecutor;
+  private final Executor callbackExecutor;
+
+  final Random random = new Random();
+
+  private volatile int delayMs = DEFAULT_DELAY_MS;
+  private volatile int variancePct = DEFAULT_VARIANCE_PCT;
+  private volatile int errorPct = DEFAULT_ERROR_PCT;
+
+  private MockRetrofit(Retrofit retrofit, ExecutorService backgroundExecutor) {
+    this.backgroundExecutor = backgroundExecutor;
+    this.callbackExecutor = retrofit.callbackExecutor();
+  }
+
+  /** Create a call which succeeds with {@code body} in its response. */
+  public <T> Call<T> newSuccessCall(T body) {
+    return newSuccessCall(Response.fakeSuccess(body));
+  }
+
+  /** Create a call which succeeds with {@code response}. */
+  public <T> Call<T> newSuccessCall(Response<T> response) {
+    return new MockCall<>(this, backgroundExecutor, callbackExecutor, response, null);
+  }
+
+  /** Create a call which fails with {@code e}.  */
+  public <T> Call<T> newFailureCall(IOException e) {
+    return new MockCall<>(this, backgroundExecutor, callbackExecutor, null, e);
+  }
+
+  /** Set the network round trip delay. */
+  public void setDelay(long amount, TimeUnit unit) {
+    if (amount < 0) {
+      throw new IllegalArgumentException("Amount must be positive value.");
+    }
+    long delayMs = unit.toMillis(amount);
+    if (delayMs > Integer.MAX_VALUE) {
+      throw new IllegalArgumentException("Delay value too large. Max: " + Integer.MAX_VALUE);
+    }
+    this.delayMs = (int) delayMs;
+  }
+
+  /** The network round trip delay. */
+  public long delay(TimeUnit unit) {
+    return MILLISECONDS.convert(delayMs, unit);
+  }
+
+  /** Set the plus-or-minus variance percentage of the network round trip delay. */
+  public void setVariancePercentage(int variancePct) {
+    if (variancePct < 0 || variancePct > 100) {
+      throw new IllegalArgumentException("Variance percentage must be between 0 and 100.");
+    }
+    this.variancePct = variancePct;
+  }
+
+  /** The plus-or-minus variance percentage of the network round trip delay. */
+  public int variancePercentage() {
+    return variancePct;
+  }
+
+  /** Set the percentage of calls to {@link #calculateIsFailure()} that return {@code true}. */
+  public void setErrorPercentage(int errorPct) {
```

Comment:
Not that I know of. I chose an integer for a few reasons:
 * You're unlikely to want higher granularity than whole percents.
 * Reading and writing to somewhere persisted is easy.
 * It's easier to present in UI, especially as whole numbers (which we do).

It's not a range of 100, but the individual channels of colors are usually presented as integers rather than floats. Only example I can think of...